device/alpaka/src/utils/thread_id.hpp

```diff
@@ -21,49 +21,50 @@
 namespace traccc::alpaka::details {
 
 /// An Alpaka thread identifier type
-template <typename Acc>
+template <typename TAcc>
 struct thread_id1 {
-    TRACCC_HOST_DEVICE explicit thread_id1(const Acc& acc) : m_acc(acc) {}
+    ALPAKA_FN_INLINE ALPAKA_FN_ACC explicit thread_id1(const TAcc* acc)
+        : m_acc(acc) {}
 
-    unsigned int inline TRACCC_HOST_DEVICE getLocalThreadId() const {
+    unsigned int inline ALPAKA_FN_ACC getLocalThreadId() const {
         return static_cast<unsigned int>(
-            ::alpaka::getIdx<::alpaka::Block, ::alpaka::Threads>(m_acc)[0u]);
+            ::alpaka::getIdx<::alpaka::Block, ::alpaka::Threads>(*m_acc)[0u]);
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getLocalThreadIdX() const {
+    unsigned int inline ALPAKA_FN_ACC getLocalThreadIdX() const {
         return getLocalThreadId();
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getGlobalThreadId() const {
+    unsigned int inline ALPAKA_FN_ACC getGlobalThreadId() const {
         return getLocalThreadId() + getBlockIdX() * getBlockDimX();
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getGlobalThreadIdX() const {
+    unsigned int inline ALPAKA_FN_ACC getGlobalThreadIdX() const {
         return getLocalThreadId() + getBlockIdX() * getBlockDimX();
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getBlockIdX() const {
+    unsigned int inline ALPAKA_FN_ACC getBlockIdX() const {
         return static_cast<unsigned int>(
-            ::alpaka::getIdx<::alpaka::Grid, ::alpaka::Blocks>(m_acc)[0u]);
+            ::alpaka::getIdx<::alpaka::Grid, ::alpaka::Blocks>(*m_acc)[0u]);
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getBlockDimX() const {
+    unsigned int inline ALPAKA_FN_ACC getBlockDimX() const {
         return static_cast<unsigned int>(
             ::alpaka::getWorkDiv<::alpaka::Block, ::alpaka::Threads>(
-                m_acc)[0u]);
+                *m_acc)[0u]);
     }
 
-    unsigned int inline TRACCC_HOST_DEVICE getGridDimX() const {
+    unsigned int inline ALPAKA_FN_ACC getGridDimX() const {
         return static_cast<unsigned int>(
-            ::alpaka::getWorkDiv<::alpaka::Grid, ::alpaka::Blocks>(m_acc)[0u]);
+            ::alpaka::getWorkDiv<::alpaka::Grid, ::alpaka::Blocks>(*m_acc)[0u]);
     }
 
     private:
-    const Acc& m_acc;
+    const TAcc* m_acc;
```

Comment:
I'm not happy about this change. 🤔

If the struct needs to be "assignable", please replace the current reference with `std::reference_wrapper<TAcc>` instead. I don't like that the struct now receives a pointer, which itself is not checked anywhere for being a `nullptr` or not.